docs/yul.rst

```diff
@@ -7,15 +7,13 @@ Yul
 .. index:: ! assembly, ! asm, ! evmasm, ! yul, julia, iulia
 
 Yul (previously also called JULIA or IULIA) is an intermediate language that can
-compile to various different backends
-(EVM 1.0, EVM 1.5 and eWASM are planned).
-Because of that, it is designed to be a usable common denominator of all three
-platforms.
-It can already be used for "inline assembly" inside Solidity and
-future versions of the Solidity compiler will even use Yul as intermediate
-language. It should also be easy to build high-level optimizer stages for Yul.
-
-In its flavour of inline-assembly, Yul can be used as a language setting
+compile bytecode output for different backends.
+
+Support for EVM 1.0, EVM 1.5 and eWASM is planned, and it is designed to be a usable common denominator of all three
+platforms. It can already be used for "inline assembly" inside Solidity and future versions of the Solidity compiler
```

Comment:
Maybe we should remove this sentence about "inline assembly" here (continuing directly with Yul as a good language for optimization and an experimental Yul optimizer being part of the Solidity compiler). Then after this we could have a new paragraph explaining the whole flavor thing:

``Yul comes in two flavours, namely untyped Yul and typed Yul.`` (naming up for discussion, but I'd prefer this over "strict" and "non-strict" to avoid confusion) ``While future versions of Solidity will use (typed) Yul as intermediate language, untyped Yul can already be used in Solidity as "strict inline assembly". This mode for inline assembly can be enabled for the command line interface using the --strict-assembly parameter. Furthermore, untyped Yul can be used as language setting for the standard-json interface.``